### PR TITLE
Duplicate Datasources

### PR DESCRIPTION
**Pull request by dehume**
Investigation for duplicate datasources. It seems that the issue was with `materialize_materialized_views` that contained a top level comment as well as column level comments. This cannot be set via the provider (mostly likely dbt) which was leading to duplicated rows in the data sources. Including `AND object_sub_id IS NULL` to the subquery removes the column level comments.

Also including acceptance tests for all data sources:
| Datasource | Status |
| --- | --- |
| cluster replica | X |
| cluster | X |
| connection | X |
| current cluster | X |
| current database | X |
| database | X |
| egress ip | X |
| index | X |
| materialized view | X |
| role | X |
| schema | X |
| secret | X |
| sink | X |
| source | X |
| table | X |
| type | X |
| view | X |

### CODE REVIEW

File: pkg/utils/ids_migration_helper_test.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ func TestTransformIdWithRegion(t *testing.T) {
 			"expected": "aws/us-east-1:u1",
 		},
 	}
-	for tc, _ := range testCases {
+	for tc := range testCases {
```
Comment (dehume) [inline pkg/utils/ids_migration_helper_test.go:29]: Unrelated, just from running format

**Review by bobbyiliev — CHANGES_REQUESTED** [review]
Seems like that the IDs migration CI is failing, but it should no longer be needed as the version has now been released.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,12 @@
 
 ## Unreleased
 
+### Features
+* Allow Avro comments (`avro_doc_type` and `avro_doc_column`) for resource `materialize_sink_kafka` [#373](https://github.com/MaterializeInc/terraform-provider-materialize/pull/373)
```
Comment (dehume) [inline CHANGELOG.md:6]: Was not included in the last PR

File: pkg/materialize/materialized_view.go [hunk 3]
```diff
@@ -114,6 +114,7 @@ var materializedViewQuery = NewBaseQuery(`
 		SELECT id, comment
 		FROM mz_internal.mz_comments
 		WHERE object_type = 'materialized-view'
+		AND object_sub_id IS NULL
```
Comment (dehume) [inline pkg/materialize/materialized_view.go:117]: This is the actual fix for the duplication issue. The majority of the PR is just testing

**Review by bobbyiliev — APPROVED** [review]
This is great! Thank you for adding all of those tests!

**Pull request merged**